### DISCUSSION TITLE
Exult 1.8.0 doesn't setup GOG U7(BG+Forge) and U7:SI+SS properly (exult.cfg issue)

### DISCUSSION THREAD

**Discussion by 2bitcounter** [exult/exult#431] [General]
Hello,

I bought GOG.com's "Ultima 7 The Complete Collection" and installed each game's installer separately for U7(BG+Forge) and U7:SI+SS. I am using Win10 64-bit.
I then installed Exult v1.8.0 and did not change any setup options in the installer (i.e. it will setup games).
It asked for the paths for both games, which were 
"C:\GOG Games\Ultima 7" and "C:\GOG Games\Ultima 7 - Serpent Isle". Note that the Exult installer did not complain about either of the paths, and I was able to complete the Exult install.
However, on running Exult, it complained about "could not find the static folder" for both games.

Here is what exult.cfg looked after installing and before any editing was done:
<config>
 <disk>
  <save_compression_level>
  1
  </save_compression_level>
 </disk>
 <video>
  <fullscreen>
  no
  </fullscreen>
  <game>
   <border>
    <red>
    0
    </red>
    <green>
    0
    </green>
    <blue>
    0
    </blue>
   </border>
   <width>
   320
   </width>
   <height>
   200
   </height>
  </game>
  <display>
   <width>
   1024
   </width>
   <height>
   768
   </height>
  </display>
  <scale>
  2
  </scale>
  <scale_method>
  Point
  </scale_method>
  <fill_mode>
  Fit
  </fill_mode>
  <fill_scaler>
  Point
  </fill_scaler>
  <highdpi>
  yes
  </highdpi>
  <share_video_settings>
  yes
  </share_video_settings>
 </video>
 <gameplay>
  <right_click_closes_gumps>
  yes
  </right_click_closes_gumps>
  <gumps_dont_pause_game>
  no
  </gumps_dont_pause_game>
  <mouse3rd>
  no
  </mouse3rd>
  <fastmouse>
  no
  </fastmouse>
  <double_click_closes_gumps>
  no
  </double_click_closes_gumps>
  <combat>
   <difficulty>
   0
   </difficulty>
   <charmDifficulty>
   normal
   </charmDifficulty>
   <mode>
   original
   </mode>
   <show_hits>
   no
   </show_hits>
  </combat>
  <step_tile_delta>
  8
  </step_tile_delta>
  <allow_right_pathfind>
  double
  </allow_right_pathfind>
  <formation>
  yes
  </formation>
  <smooth_scrolling>
  0
  </smooth_scrolling>
  <alternate_drop>
  no
  </alternate_drop>
  <allow_autonotes>
  no
  </allow_autonotes>
  <scroll_with_mouse>
  yes
  </scroll_with_mouse>
  <extended_intro>
  yes
  </extended_intro>
 </gameplay>
 <audio>
  <disablepause>
  no
  </disablepause>
  <enabled>
  yes
  </enabled>
  <midi>
   <enabled>
   yes
   </enabled>
   <convert>
   gm
   </convert>
   <use_oggs>
   no
   </use_oggs>
   <driver>
   gm
   </driver>
  </midi>
 </audio>
 <touch>
  <item_menu>
  no
  </item_menu>
  <dpad_location>
  no
  </dpad_location>
  <touch_pathfind>
  no
  </touch_pathfind>
 </touch>
 <shortcutbar>
  <use_shortcutbar>
  no
  </use_shortcutbar>
  <use_outline_color>
  no
  </use_outline_color>
  <hide_missing_items>
  yes
  </hide_missing_items>
 </shortcutbar>
</config>

In case the above copy & paste looks like garbage I'm also attaching the original non-working .cfg file to this message.

`You'll notice that despite the installer being pointed to the GOG file paths there is no <blackgate> or  <serpentisle> with the relevant <path> sections under <games>`

I am not sure why this happened, as the GOG path in Exult docs section 11.9 matches my path for both games:
_Right now the only way to get at the game files is to run the installer on Windows (or in WINE if you are not a Windows user). After the installation you will find the game files in C:\GOG Games\Ultima 7 and C:\GOG Games\Ultima 7 - Serpent Isle.
After installation you can just run our installer and when asked where the games are installed enter or browse to the above paths.
Unfortunately GOG changes the download of the games for some unknown reason again and again. So our instructions are sometimes out of date. These ones are for the latest GOG release at the time of writing. If it doesn't work, please visit our forum and tell us._ 

So I'm telling you ;-)

**The way I got both games to work was by replacing the contents of my exult.cfg file in %APPDATA%/Local/Exult with the one from the following link:
[https://pastebin.com/zR6wNgHk](https://pastebin.com/zR6wNgHk)
and then editing exult.cfg and:**
```
changing the path for <blackgate> to "C:\GOG Games\Ultima 7"
changing the path for <serpentisle> to "C:\GOG Games\Ultima 7 - Serpent Isle"
```

I do not know why the Exult installer is not setting up the paths or other options in exult.cfg correctly (there's a lot of stuff in the pastebin link that was not in my .cfg after installing).

But it was worth this hassle - Ultima looks great with the antialias filtering and much better than the original DOS version (running under DOSBox).

I have documented this so it can be fixed, and also to save someone who reads this in the future trying to figure out why it ain't working.

[exult - Original Copy.zip](https://github.com/exult/exult/files/13934483/exult.-.Original.Copy.zip)

**Comment by wench** [comment]
Creating an issue and self Assigning as I had the installer fail to correctly save the cfg file in the past so with this report I am certain that there is something that needs fixing. move all discussion to issue #432
